contrib/systemd-with-workers/system/matrix-synapse-homeserver.service

```diff
@@ -0,0 +1,17 @@
+[Unit]
+Description=Synapse Matrix Homeserver
+
+[Service]
+Type=forking
```

Comment:
I'd recommend using `Type=simple` (and making sure that synapse is not configured to `daemonize`). Systemd is better at managing daemonization than synapse, and it will solve your problem with the pid files.

Reply:
Does this work with workers?

All worker tutorials I have found so far seem to suggest that daemonizing is required. Or do you mean only non-daemonize the homeserver app and still daemonize the workers?

Reply:
Nevermind. I just tested it and it works. Will change everything to a non-forking setup.